lib/Sema/CSGen.cpp

```diff
@@ -2422,7 +2427,9 @@ namespace {
           // Resolve the parent type.
           Type parentType =
             resolveTypeReferenceInExpression(enumPattern->getParentType());
-          parentType = CS.openUnboundGenericType(parentType, locator);
+          parentType = CS.openUnboundGenericType(
+              parentType,
+              locator.withPathElement(LocatorPathElt::PatternMatch(pattern)));
```

Comment:
I think we could use `ParentType` locator element for this transformation.

Reply:
Later one where we add a member constraint I think it should either end on regular `Member` locator or a new one specific for patterns...